app/webpacker/components/RegistrationsV2/Register/Processing.jsx

```diff
@@ -1,30 +1,13 @@
-import { useQuery } from '@tanstack/react-query';
-import React, { useEffect, useState } from 'react';
+import React from 'react';
 import { Message, Modal } from 'semantic-ui-react';
 import I18n from '../../../lib/i18n';
-import pollRegistrations from '../api/registration/get/poll_registrations';
+import { useRegistration } from '../lib/RegistrationProvider';
 
-const REFETCH_INTERVAL = 3000;
+export default function Processing() {
+  const { pollCounter, queueCount } = useRegistration();
 
-export default function Processing({ competitionInfo, user, onProcessingComplete }) {
-  const [pollCounter, setPollCounter] = useState(0);
-
-  const { data } = useQuery({
-    queryKey: ['registration-status-polling', user.id, competitionInfo.id],
-    queryFn: async () => pollRegistrations(user.id, competitionInfo),
-    refetchInterval: REFETCH_INTERVAL,
-    onSuccess: () => {
-      setPollCounter(pollCounter + 1);
-    },
-  });
-  useEffect(() => {
-    const processingComplete = data && !data.processing;
-    if (processingComplete) {
-      onProcessingComplete();
-    }
-  }, [data, onProcessingComplete]);
   return (
-    <Modal open={data?.status?.competing !== 'pending' || !data?.processing} dimmer="blurring">
+    <Modal open dimmer="blurring">
```

Comment:
Why is it suddenly viable to keep the modal always open?

Reply:
because of 
```
{isPolling && (
        <Processing
          competitionInfo={competitionInfo}
          user={user}
        />
      )}
```